### PR TITLE
doc: document runtime config

### PR DESCRIPTION
**Pull request by magiconair**
Moving the previous `agent/config.go` documentation to `agent/config/runtime.go`.

### CODE REVIEW

File: agent/config/runtime.go [hunk 1]
```diff
@@ -48,118 +55,492 @@ type RuntimeConfig struct {
...
+	// hcl: acl_down_policy = ("allow"|"deny"|"extend-cache")
+	ACLDownPolicy string
+
+	// ACLEnforceVersion8 is used to gate a set of ACL policy features that
+	// are opt-in prior to Consul 0.8 and opt-out in Consul 0.8 and later.
+	//
+	// hcl: acl_enforce_version_8 = (true|false)
+	ACLEnforceVersion8 bool
+
+	// ACLEnableKeyListPolicy ???
```
Comment (slackpad) [inline agent/config/runtime.go]: ACLEnableKeyListPolicy is used to opt-in to the "list" policy added to KV ACLs in Consul 1.0. See https://www.consul.io/docs/guides/acl.html#list-policy-for-keys for more details.

**Comment by magiconair** [comment]
I think it is a good idea to structure the `RuntimeConfig` a bit more. The TLS related options are in different places. I'll add a separate PR once I'm done with the documentation.

File: agent/config/runtime.go [hunk 2]
```diff
@@ -48,160 +55,1173 @@ type RuntimeConfig struct {
...
-	SyslogFacility              string
-	TLSCipherSuites             []uint16
-	TLSMinVersion               string
+
+	// Node name is the name we use to advertise. Defaults to hostname.
+	//
+	// NodeName is exposed via /v1/agent/self from here and
+	// used in lots of places like CLI commands. Treat this as an interface
+	// that must be stable.
+	// todo(fs): do we need this warning? RuntimeConfig was meant to be refactorable.
```
Comment (slackpad) [inline agent/config/runtime.go]: This comment doesn't apply anymore now that we expose a `Config` member in /v1/agent/self that's not based on this structure. You can delete it.

File: agent/config/runtime.go [hunk 3]
```diff
@@ -48,160 +55,1173 @@ type RuntimeConfig struct {
...
+
+	// Bootstrap is used to bring up the first Consul server, and
+	// permits that node to elect itself leader
+	//
+	// hcl: bootstrap = (true|false)
+	// flag: -bootstrap
+	Bootstrap bool
+
+	// BootstrapExpect tries to automatically bootstrap the Consul cluster,
+	// by withholding peers until enough servers join.
```
Comment (slackpad) [inline agent/config/runtime.go]: "by having servers wait to bootstrap until enough servers join, and then performing the bootstrap process automatically. They will disable their automatic bootstrap process if they detect any servers that are part of an existing cluster, so it's safe to leave this set to a non-zero value."

**Comment by magiconair** [comment]
Reverted the rebased commit and re-merged as a proper merge commit.

**Pull request merged**